### PR TITLE
mm/mm_heap: Fix warnings included by #266

### PR DESCRIPTION
**Pull request by Ouss4**
@yamt Please, make sure no other issue was included by this PR.

### CODE REVIEW

File: mm/mm_heap/mm_mallinfo.c [hunk 1]
```diff
@@ -105,9 +106,10 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
             }
           else
             {
-              FAR struct mm_freenode_s *fnode;
+#ifdef CONFIG_DEBUG_ASSERTIONS
+              FAR struct mm_freenode_s *fnode = (FAR void *)node;
+#endif
               DEBUGASSERT(node->size >= SIZEOF_MM_FREENODE);
-              fnode = (FAR void *)node;
               DEBUGASSERT(fnode->blink->flink == fnode);
```
Comment (patacongo) [inline mm/mm_heap/mm_mallinfo.c]: This assertion can't be right.  If line 110 is removed, then fnode has no value.

Comment (Ouss4) [reply mm/mm_heap/mm_mallinfo.c]: You mean it won't have a value for DEBUGASSERT?
But DEBBUGASSERT is only active when assertions are enabled.

Comment (patacongo) [reply mm/mm_heap/mm_mallinfo.c]: Then don't you get a warning about fnode being unused?

Comment (patacongo) [reply mm/mm_heap/mm_mallinfo.c]: Perhaps including all of lines 108 though 117 in #ifdef CONFIG_DEBUG_ASSERTIONS would eliminate warnings about the unused variable. when CONFIG_DEBUG_ASSERTIONS is not defined.

Comment (Ouss4) [reply mm/mm_heap/mm_mallinfo.c]: This is the original warning (among 2 others) that this PR fixes.
When assertions are disabled `fnode` and `prev` were not used.

Comment (patacongo) [reply mm/mm_heap/mm_mallinfo.c]: Oh, I just noticed that fnodes is doubly initialized.  That is why you can remove the second initialization.  Sorry, I didn't follow that corrected.

This should still cause a warning about an unused variable and still should be enclosed in conditional logic.

**Pull request merged**